provider/dyn/dyn.go

```diff
@@ -145,7 +154,14 @@ type ZonePublishResponse struct {
 // NewDynProvider initializes a new Dyn Provider.
 func NewDynProvider(config DynConfig) (provider.Provider, error) {
 	return &dynProviderState{
```

Comment:
Please create a separate PR for dyn provider, I'd like to keep the PR as small as possible